server/http/src/service.rs

```diff
@@ -272,9 +272,41 @@ where
         let query = GraphQLRequest::new(body, schema, network).compat().await;
 
         let result = match query {
-            Ok(query) => graph::spawn_blocking_allow_panic(service.graphql_runner.run_query(query))
-                .await
-                .unwrap(),
+            Ok(query) => {
+                // Not great that we serialize the query here only for the panic case.
+                let query_text = query.query_text();
+                let variables_text = query.variables_text();
```

Comment:
This to me is a big problem as we will wind up formatting each query twice (once here, once for logging) Could we have `execute_query` add the query text it already has on unwind and just log that?

Alternatively, we could construct the `execution::Query` here already and pass it into the runner, though that might be a pretty intrusive change

Reply:
It's unfortunate but generally the cost of processing of query inputs is small compared to processing large query outputs. But I'll see about moving `execution::Query` into `graph` so we can construct it directly.